insights/parsers/vgdisplay.py

```diff
@@ -1,89 +1,184 @@
+        LV Name                Root
+        VG Name                RHEL7CSB
+        LV Size                29.30 GiB
+        ...
+
+        --- Physical volumes ---
+        PV Name               /dev/mapper/luks-96c66446-77fd-4431-9508-f6912bd84194
+        PV UUID               EfWV9V-03CX-E6zc-JkMw-yQae-wdzp-Je1KUn
+        PV Status             allocatable
+        Total PE / Free PE    118466 / 4036
+
+    Lines containing 'Couldn't find device with uuid' and 'missing physical
+    volume' are stored in a ``debug_info`` property, which is also available
+    in the ``debug_info`` key in the ``data`` dictionary.
+
+    Examples:
+        >>> vg_info = shared[VgDisplay]
+        >>> len(vg_info.vg_list)
+        1
+        >>> vgdata = vg_info.vg_list[0]
+        >>> vgdata['VG Name']
+        'RHEL7CSB'
+        >>> vgdata['VG Size']
+        '462.76 GiB'
+        >>> 'Logical Volumes' in vgdata
+        True
+        >>> lvs = vgdata['Logical Volumes']
+        >>> type(lvs)
+        dict
+        >>> lvs.keys()  # Note - keyed by device name
+        ['/dev/RHEL7CSB/Root']
+        >>> lvs['/dev/RHEL7CSB/Root']['LV Name']
+        'Root'
+        >>> lvs['/dev/RHEL7CSB/Root']['LV Size']
+        '29.30 GiB'
+        >>> 'Physical Volumes' in vgdata
+        True
+        >>> vgdata['Physical Volumes'].keys()
+        ['/dev/mapper/luks-96c66446-77fd-4431-9508-f6912bd84194']
+        >>> vgdata['Physical Volumes']['/dev/mapper/luks-96c66446-77fd-4431-9508-f6912bd84194']['PV UUID']
+        'EfWV9V-03CX-E6zc-JkMw-yQae-wdzp-Je1KUn'
+        >>> vg_info.debug_info
+        ["Couldn't find device with uuid VVLmw8-e2AA-ECfW-wDPl-Vnaa-0wW1-utv7tV."]
     """
 
+    _FILTER_INFO = [
+        "Couldn't find device with uuid",
+        "physical volumes missing",
+    ]
+
+    VGDISPLAY = """
+Couldn't find device with uuid VVLmw8-e2AA-ECfW-wDPl-Vnaa-0wW1-utv7tV.
+  There are 1 physical volumes missing.
+    Couldn't find device with uuid VVLmw8-e2AA-ECfW-wDPl-Vnaa-0wW1-utv7tV.
+  There are 1 physical volumes missing.
+  --- Volume group ---
+  VG Name               rhel_hp-dl160g8-3
+  Format                lvm2
+  System ID
+  Metadata Areas        1
+  Metadata Sequence No  5
+  VG Access             read/write
+  Total PE              119109
+  Alloc PE / Size       119098 / 465.23 GiB
+  Free  PE / Size       11 / 44.00 MiB
+  VG UUID               by0Dl3-0lpB-MxEz-f6GO-9LYO-YRAQ-GufNZD
+
+  VG Name               rhel_hp-dl260g7-4
+  Format                lvm2
+  System ID
+  VG Access             read/write
+  Alloc PE / Size       119098 / 465.23 GiB
+  Free  PE / Size       11 / 44.00 MiB
+  VG UUID               by0Dl3-0lpB-MxEz-f6GO-9LYO-YRAQ-GufNZN
+"""
+
     def parse_content(self, content):
         self.data = {}
         vginfo_dict_list = []
-        vginfo_dict = {}
-        debug_info = []
-        for line in get_active_lines(content):
+        self.debug_info = []
+        self.data['debug_info'] = self.debug_info
```

Comment:
Is the `data` attribute used anywhere outside of this method?  Could it be removed, or be a local var?